src/ralph/backends/data/fs.py

```diff
@@ -0,0 +1,326 @@
+"""FileSystem data backend for Ralph."""
+
+import json
+import logging
+import os
+from datetime import datetime, timezone
+from io import IOBase
+from itertools import chain
+from pathlib import Path
+from typing import IO, Iterable, Iterator, Union
+from uuid import uuid4
+
+from ralph.backends.data.base import (
+    BaseDataBackend,
+    BaseDataBackendSettings,
+    BaseOperationType,
+    BaseQuery,
+    DataBackendStatus,
+    enforce_query_checks,
+)
+from ralph.backends.mixins import HistoryMixin
+from ralph.conf import BaseSettingsConfig
+from ralph.exceptions import BackendException, BackendParameterException
+from ralph.utils import now
+
+logger = logging.getLogger(__name__)
+
+
+class FSDataBackendSettings(BaseDataBackendSettings):
+    """Represents the FileSystem data backend default configuration.
+
+    Attributes:
+        DEFAULT_CHUNK_SIZE (int): The default chunk size for reading files.
+        DEFAULT_DIRECTORY_PATH (str or Path): The default target directory path where to
+            perform list, read and write operations.
+        DEFAULT_QUERY_STRING (str): The default query string to match files for the read
+            operation.
+        LOCALE_ENCODING (str): The encoding used for writing dictionaries to files.
+    """
+
+    class Config(BaseSettingsConfig):
+        """Pydantic Configuration."""
+
+        env_prefix = "RALPH_BACKENDS__DATA__FS__"
+
+    DEFAULT_CHUNK_SIZE: int = 4096
+    DEFAULT_DIRECTORY_PATH: Path = Path(".")
+    DEFAULT_QUERY_STRING: str = "*"
+    LOCALE_ENCODING: str = "utf8"
+
+
+class FSDataBackend(HistoryMixin, BaseDataBackend):
+    """FileSystem data backend."""
+
+    name = "fs"
+    default_operation_type = BaseOperationType.CREATE
+    settings_class = FSDataBackendSettings
+
+    def __init__(self, settings: settings_class = None):
+        """Creates the default target directory if it does not exist."""
+        settings = settings if settings else self.settings_class()
+        self.default_chunk_size = settings.DEFAULT_CHUNK_SIZE
+        self.default_directory = settings.DEFAULT_DIRECTORY_PATH
+        self.default_query_string = settings.DEFAULT_QUERY_STRING
+        self.locale_encoding = settings.LOCALE_ENCODING
+
+        if not self.default_directory.is_dir():
+            msg = "Default directory doesn't exist, creating: %s"
+            logger.info(msg, self.default_directory)
+            self.default_directory.mkdir(parents=True)
```

Comment:
You are right) This part is only executed when the default directory is not a directory (as this value is defined in the settings, it could point to any location).